fuseimg/datasets/cmmd.py

```diff
@@ -0,0 +1,117 @@
+    # put on the folliwing in the main folder  - 
+    # 1. CMMD_clinicaldata_revision.csv which is a converted version of CMMD_clinicaldata_revision.xlsx 
+    # 2. folder named CMMD which is the downloaded data folder
+    """
+    # bump whenever the static pipeline modified
+    CMMD_DATASET_VER = 0
+
+    @staticmethod
+    def static_pipeline(data_dir: str,data_source : pd.DataFrame) -> PipelineDefault:
+        """
+        Get suggested static pipeline (which will be cached), typically loading the data plus design choices that we won't experiment with.
+        :param data_path: path to original kits21 data (can be downloaded by KITS21.download())
+        """
+        static_pipeline = PipelineDefault("static", [
+            
+            (OpReadDataframe(data_source,key_column = None , columns_to_extract = ['file','classification'] , rename_columns=dict(file="data.input.img_path",classification="data.gt.classification")), dict()), # will save image and seg path to "data.input.img_path", "data.gt.seg_path" 
+            (OpLoadDicom(data_dir), dict(key_in="data.input.img_path", key_out="data.input.img", format="nib")),
+            (OpFlipBrightSideOnLeft2D(), dict(key="data.input.img")),
+            (OpFindBiggestNonEmptyBbox2D(), dict(key="data.input.img")),
+            (OpNormalizeAgainstSelfImpl(), dict(key="data.input.img")),
+            (OpResizeAndPad2D(), dict(key="data.input.img", resize_to=(2200, 1200), padding=(60, 60))),
+            ])
+        return static_pipeline
+
+    @staticmethod
+    def dynamic_pipeline():
+        """
+        Get suggested dynamic pipeline. including pre-processing that might be modified and augmentation operations. 
+        """
+        dynamic_pipeline = PipelineDefault("dynamic", [
+            (OpToTensor(), dict(key="data.input.img",dtype=torch.float32)),
+            (OpSample(OpAugAffine2D()), dict(
+                            key="data.input.img",
+                            rotate=Uniform(-30.0,30.0),        
+                            scale=Uniform(0.9, 1.1),
+                            flip=(RandBool(0.3), RandBool(0.5)),
+                            translate=(RandInt(-10, 10), RandInt(-10, 10))
+                        )),
+            (OpSample(OpAugColor()), dict(
+                        key="data.input.img",
+                        gamma=Uniform(0.9, 1.1), 
+                        contrast=Uniform(0.85, 1.15),
+                        mul =  Uniform(0.95, 1.05),
+                        add=Uniform(-0.06, 0.06)
+                    )),
+            (OpLambda(partial(torch.unsqueeze, dim=0)), dict(key="data.input.img")), 
+            
+        ])
+        return dynamic_pipeline
+
+    @staticmethod
+    def create_dataset_partition(phase : str,
+                                data_dir: str,
+                                data_source : pd.DataFrame,
```

Comment:
Why do you need data_source and specific_ids? it's the same, right?
Can you rename the function name to dataset?

Reply:
I used for debug, because of this one problematic image.
I can remove it and only use data_source